### PR TITLE
Expose theme property to metabase provider in embedding SDK

### LINKED ISSUE #41948: Expose basic theme configuration to the MetabaseProvider in embedding SDK

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/41948

### Description

Adds an optional `theme` option to `MetabaseProvider`, allowing SDK users to add their own Mantine themes. The theme is then merged with Metabase's default theme overrides in our `ThemeProvider`.

### How to verify

See the `basic-theming` branch in the demo app. This branch does the following:

1. Pass in Mantine's theme options to the `MetabaseProvider` in the demo app.
2. Modify the `colors.brand` or `colors.text-dark` theme option in the demo app.
3. Color in `colors.brand` should be applied to the Mantine loader components.

### Screenshot

Notice the loader is now pink instead of light blue.

CleanShot 2567-04-30 at 23 35 21@2x

### Notes

- We should review how we write tests for this as well. For example, verifying that adding `colors.brand` will change the rendered button's color. I'll work on the tests next.

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/types/theme.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { MantineThemeOverride } from "@mantine/core";
+
+export type MetabaseTheme = MantineThemeOverride & {
+  other: MetabaseThemeOptions;
+};
+
+/**
+ * Theme options for Metabase components and visualizations.
+ */
+interface MetabaseThemeOptions {}
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/types/theme.ts:10]: This is a stub for now - more custom theme options will be added as they arise, and we should export this type later as well.

File: frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
 
 interface ThemeProviderProps {
   children: ReactNode;
+
+  /**
+   * Extend Metabase's theme overrides.
+   * This is primarily used in the React embedding SDK
+   * to allow SDK users to customize the theme.
+   */
+  theme?: MantineThemeOverride;
```
Comment (WiNloSt) [inline frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx:19]: should we use `MetabaseTheme` like we pass it in `enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx` for consistency?

**Review by WiNloSt — APPROVED** [review]
Looks good to me.

File: frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx (hunk 2 above)
Comment (WiNloSt) [reply frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx:19]: Fair enough. Also, I thought there's a way to type `theme.other`, but it doesn't look like Mantine allow that either.

Comment (heypoom) [reply frontend/src/metabase/ui/components/theme/ThemeProvider/ThemeProvider.tsx:19]: @WiNloSt in the Mantine V6 docs, they mentioned we are supposed to use declaration merging to define this. It's not in the Mantine V7 docs, but I think this method still works as the type is still `MantineThemeOther`.

CleanShot 2567-05-02 at 16 01 24@2x

**Pull request merged**